### PR TITLE
eBUS Maintenance Update 1

### PR DESCRIPTION
**Pull request by csowada**
This updates contains several enhancements. Now this binding is prepared for Vaillant devices.

- New parameter "set" allows to set a value
- Improved sender timings
- Enhanced Wolf and Vaillant configurations
- Add CSV support to record raw telegrams
- Smaller bugfixes

### CODE REVIEW

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/internal/connection/AbstractEBusConnector.java [hunk 1]
```diff
@@ -197,16 +156,19 @@ public void run() {
...
 
 				} else {
-					int read = inputStream.read();
+					read = inputStream.read();
+					lastReceiveTime = System.nanoTime();
+					bufferSize = inputStream.available();
+					
 					if(read == -1) {
-						logger.error("eBus read timeout occured, ignore it currently!");
+						logger.info("eBUS read timeout occured, no data on bus ...");
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/internal/connection/AbstractEBusConnector.java]: please decrease severity to debug

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+public class EBusJsonConfTool {
+
+	static String ALLOWED_LEVEL_1_PARAMS[] = {"comment", "device", "id", "class", "command", "data", "values", "computed_values", "debug"};
+
+
+	/**
+	 * @param args
+	 */
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java]: remove comment

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+		PrintStream out = null;
+		
+		PrintStream mainOut = new PrintStream(new File(currentDir, "docs/json-configs.md"));
+		mainOut.println("# JSON configuration files");
+		mainOut.println();
+		mainOut.println("This is an automatic created list of all included configuration files.");
+		mainOut.println();
+		
+		EBusConfigurationProvider ebuscfg = new EBusConfigurationProvider();
+//		ebuscfg.loadConfigurationFile(url);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java]: remove commented code

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+
+			
+		}
+
+		mainOut.println();
+		mainOut.flush();
+		mainOut.close();
+
+		//        List<Map<String, ?>> configurationMap = mapper.readValue(new File("C:\\Users\\csowada\\Downloads\\vaillant-bai00-configuration.json"), List.class);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java]: remove commented code

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+
+
+		
+
+		checkConfiguration(configurationMap);
+//
+//		writeMarkdownIdTable(configurationMap, out);
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java]: remove commented code

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+	}
+
+}
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/EBusJsonConfTool.java]: add newline

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/tests/TestClient.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
+package org.openhab.binding.ebus.tools.tests;
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/tests/TestClient.java]: add license header and documentation

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/tests/TestClient2.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
+package org.openhab.binding.ebus.tools.tests;
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/tests/TestClient2.java]: add license header and documentation

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/tests/TestClient3.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
+package org.openhab.binding.ebus.tools.tests;
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/tests/TestClient3.java]: add license header and documentation

File: bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/tests/TestClient3.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+//		Pattern pattern = Pattern.compile("(\\([0-9A-Z]{2}\\))");
+//		String replaceAll = StringUtils.strip(pattern.matcher(a).replaceAll(""));
+//		
+////		String replaceAll = a.replaceAll("(\\([0-9A-Z]{2}\\))", "--");
+//		String replaceAll2 = a.replaceAll("(\\(|\\))", "");
+//		System.out.println("TestClient3.main() " + replaceAll);
+//		System.out.println("TestClient3.main() " + replaceAll2);
+//		
+//		if(true)
+//			return;
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/java/org/openhab/binding/ebus/tools/tests/TestClient3.java]: remove commented code

File: bundles/binding/org.openhab.binding.ebus/src/main/resources/.gitignore [hunk 11]
```diff
@@ -0,0 +1 @@
+/wolf-log.csv
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/resources/.gitignore]: better add to .gitignore on bundle level

File: bundles/binding/org.openhab.binding.ebus/src/main/resources/dev-notices.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+---                 | ---                 | ---      | ---    | ---    | ---
+Burner Control 1    | Feuerungsautomat 1  | GBC-e    | 0x03   | 0x08   | Feuerungsautomat 1
+Burner Automat 1    | Bedienmodul 1       | BM2      | 0x30   | 0x35   | Heizkreisregler 1
+Loop 1              | Regelungsplatine 1  | HCM-2    | 0x50   |        | Mischer 1 (Slave only)
+Heater Control      | Heizungsregler      | HCM-2?   | 0xF1   | (0xF6) | Heizungsregler (Hauptregler?)
+Solar 1             | Solarregler         | SM1      | 0x71   | 0x76   | Heizungsregler (Solarregler)
+ISM7i               | ISM7i               | ISM7i    | 0xFF   | 0x05   | PC
+Room Control 1      |                     |          |        |        |
+Remote Actuator 1   |                     |          |        |        |
+PC 1                |                     |          |        |        |
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.ebus/src/main/resources/dev-notices.md]: either translate to english or move to wiki

**Comment by teichsta** [comment]
Hi @csowada thanks for this enhancement of the binding. Any chance to extract the fixes into a separate in order to cherry-pick them into 1.7.1? Also please squash your commits into after you've worked through my (tiny) comments. Best, Thomas E.-E.

**Comment by csowada** [comment]
Hi @teichsta,
thank you for your fast review. I will change the files you commented. I'll also check the included fixes for a seperate 1.7.1 pull request. But I think it will be to complex to extract them.

**Comment by teichsta** [comment]
>  But I think it will be to complex to extract them.

no worries, just wanted to have that clarified …

**Comment by csowada** [comment]
I think I should start a new pull request, because now it is a bit dirty with all the useless commits.
Sorry but I'm new to Git.
Here is the new clean branch: https://github.com/openhab/openhab/compare/master...csowada:ebus-update1b

**Comment by teichsta** [comment]
thanks could you please create another PR containing your branch? Thanks, Thomas E.-E.

**Comment by csowada** [comment]
Closed, replaced by PR https://github.com/openhab/openhab/pull/2981

**Pull request closed without merging**